alacritty/src/renderer/mod.rs

```diff
@@ -1573,22 +1583,42 @@ impl Atlas {
         let offset_x = self.row_extent;
         let height = glyph.height as i32;
         let width = glyph.width as i32;
+        let colored;
 
         unsafe {
             gl::BindTexture(gl::TEXTURE_2D, self.id);
 
             // Load data into OpenGL
-            gl::TexSubImage2D(
-                gl::TEXTURE_2D,
-                0,
-                offset_x,
-                offset_y,
-                width,
-                height,
-                gl::RGB,
-                gl::UNSIGNED_BYTE,
-                glyph.buf.as_ptr() as *const _,
-            );
+            match &glyph.buf {
+                BitmapBuffer::RGB(buf) => {
+                    gl::TexSubImage2D(
+                        gl::TEXTURE_2D,
+                        0,
+                        offset_x,
+                        offset_y,
+                        width,
+                        height,
+                        gl::RGB,
+                        gl::UNSIGNED_BYTE,
+                        buf.as_ptr() as *const _,
+                    );
+                    colored = false;
+                },
+                BitmapBuffer::RGBA(buf) => {
+                    gl::TexSubImage2D(
+                        gl::TEXTURE_2D,
+                        0,
+                        offset_x,
+                        offset_y,
+                        width,
+                        height,
+                        gl::RGBA,
+                        gl::UNSIGNED_BYTE,
+                        buf.as_ptr() as *const _,
+                    );
+                    colored = true;
+                },
+            }
```

Comment:
Github didn't let me add any more comments to my existing PR because you pushed a commit I think, so I've added this separately:

I think we could remove duplication a lot by changing to something more like this:

```rust
let (format, buf) = match glyph.buf {
    RGB(buf) => (gl::RGB, buf),
    RGBA(buf) => (gl::RGBA, buf),
};
gl::TexSubImage2D(format, buffer);
```